### PR TITLE
 terminal times and some more custom sounds (swap stonk sounds & term completion sounds)

### PR DESCRIPTION
**Pull request by SubAt0m1c**
blaze solver now sets blaze puzzle to complete when its complete made highlight render use renderBoundingBox so it doesnt look choppy

### CODE REVIEW

**Comment by odtheking** [comment]
I think swap sound is streching it besides that prob fine

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt [hunk 1]
```diff
@@ -35,17 +48,29 @@ object TerminalSounds : Module(
...
+                !clickSounds
             ) return
             playTerminalSound()
             event.isCanceled = true
         }
     }
 
+    @SubscribeEvent
+    fun onTermComplete(event: TerminalSolvedEvent) {
+        if (event.type == TerminalTypes.NONE || mc.currentScreen is TermSimGui || event.playerName != mc.thePlayer?.name.noControlCodes || !completeSounds) return
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt:61]: mc.theplayer.name will never have color codes

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt [hunk 2]
```diff
@@ -39,4 +46,59 @@ object TerminalTimes : Module(
...
+    private var sectionTimer = 0L
+    private val times = mutableListOf<Long>()
+
+    @SubscribeEvent
+    fun onMessage(event: ClientChatReceivedEvent) {
+        if (event.message.unformattedText.noControlCodes.matches(terminalCompleteRegex) && terminalSplits) event.isCanceled = true
+    }
+
+    init {
+        onMessage("The gate has been destroyed!", false, { terminalSplits }) {
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt:64]: need to add enabled as well for onmessage if u are overriding the default

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt [hunk 3]
```diff
@@ -39,4 +46,59 @@ object TerminalTimes : Module(
...
+            if (completed.first == completed.second) resetSection()
+            else gateBlown = true
+        }
+
+        onMessage("[BOSS] Goldor: Who dares trespass into my domain?", false, { terminalSplits }) {
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt:69]: same as one above

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt [hunk 4]
```diff
@@ -39,4 +46,59 @@ object TerminalTimes : Module(
...
+            resetSection(true)
+        }
+
+        onMessage(terminalCompleteRegex, { terminalSplits }) {
+            val matchResult = terminalCompleteRegex.find(it)?.groups ?: return@onMessage
+            val complete = Pair(matchResult[4]?.value?.toIntOrNull() ?: return@onMessage, matchResult[5]?.value?.toIntOrNull() ?: return@onMessage)
```
Comment (odtheking) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt:75]: use destructured to define all the groups from the matchresult

**Comment by SubAt0m1c** [comment]
if you want me to remove swap sounds ping me

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/TickTimers.kt [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            goldorTickTime = -1
+            goldorStartTime = -1
+        }
+
+        onMessage(Regex("\\[BOSS] Storm: I should have known that I stood no chance\\.")) {
+            if (goldorHud.enabled) goldorStartTime = 104
+            if (stormHud.enabled) padTickTime = 20
+        }
+
+        onMessage(Regex("\\[BOSS] Storm: I should have known that I stood no chance\\."), { enabled && stormHud.enabled }) { padTickTime = -1 }
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/TickTimers.kt:72]: What

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/TickTimers.kt [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+        onWorldLoad {
+            if (necronHud.enabled) necronTime = -1
+            if (goldorHud.enabled) {
+                goldorTickTime = -1
+                goldorStartTime = -1
+            }
+            if (stormHud.enabled) padTickTime = -1
+        }
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/TickTimers.kt:81]: Do u really need the if checks here?

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt [hunk 7]
```diff
@@ -22,6 +27,8 @@ object TerminalTimes : Module(
         modMessage("§6Terminal PBs §fhave been reset.")
     }
 
+    private val terminalSplits: Boolean by BooleanSetting("Terminal Splits", default = true, description = "Adds the time when a term was completed to its message, and sends the total term time after terms are done.")
+
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt:31]: Doesnt this fit way better in the splits module?

**Review by freebonsai — CHANGES_REQUESTED** [review]
Looks mostly fine, i commented on some things.
I didnt read the whole pad timer stuff but i am not sure if you did what i meant, i wanted a timer like goldor timer that resets again and again that tells u when pad will actually descend from u standing on it.

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/TickTimers.kt (hunk 5 above)
Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/TickTimers.kt:72]: this just turns off the timer when storm is done

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt (hunk 7 above)
Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalTimes.kt:31]: I dont think so. It edits existing chat messages to include more info and would be rather bulky if every segment was split into a new line.

**Comment by SubAt0m1c** [comment]
this is just whatever soshimee has by the same module name. If thats what youre basing the idea off, then this should do exactly that.

File: odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/TickTimers.kt (hunk 5 above)
Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/TickTimers.kt:71]: oh wait

**Review by freebonsai — APPROVED** [review]

**Pull request merged**